### ISSUE TITLE
Problem normalizing my .vcf files 

### ISSUE THREAD

**Issue by klmartinez** [samtools/bcftools#1062]
I am trying to normalize my genome .vcf files so that I can succesffuly merge them. I receive errors when I try to merge my genome .vcf files right away, and have found that I had to normalize my exome .vcf files in order to merge them.

I use the following pattern to normalize a single genome .vcf file:

`bcftools norm -m-any 17-10687.vcf.gz | bcftools norm -Oz --check-ref w -f  /human_g1k_v37.fasta > norm/17-10687_norm.vcf.gz`

But received the following errors:

```
bcftools: vcfnorm.c:418: split_info_numeric: Assertion `ret==src->n_allele-1' failed.
[W::vcf_parse] INFO 'gnom' is not defined in the header, assuming Type=String
[vcf.c:1252 bcf_write] Broken VCF record, the number of columns at 6:30953596 does not match the number of samples (0 vs 1).
Lines   total/split/realigned/skipped:  1703616/0/8370/0
```

When I try to normalize my other .vcf files, I receive a similar output but indicating a "Broken VCF record" at a different spot. 

Are there any suggestions on how I can fix this?

**Comment by pd3** [comment]
The error message suggests that the `INFO/gnom` annotation is not defined in the header. Since you requested multiallelic records to be broken down, this information is needed to properly handle tags defined as `Number=R,A` in the header. Fill the missing information with `bcftools reheader` and it should work. (You'll either need to provide the complete header to the `reheader` command or remove the undefined tags with `annotate -x`.)

**Comment by klmartinez** [comment]
Thank you for your reply. 

I think I am still a bit confused. Do you mean I need to fill in missing information in the meta-information lines for `##INFO=<ID=gnomAD`? I have 19 `INFO/gnom` meta-information lines, but all of fields are filled including `Number`, `Type `and `Description`. 

For example, one `INFO/gnom` meta-information line looks like the following: 
`##INFO=<ID=gnomAD_combined_AF_FIN,Number=A,Type=Float,Description="Allele Frequency among Finnish genotypes, for each ALT allele, in the same order as listed">`

Is there a downside to just removing undefined tags like you suggested? Will I be losing information? Would I just go about doing this with `bcftools annotate -x INFO file.vcf.gz`?

Sorry for all my questions, this is all very new to me and I appreciate any suggestions anyone might have.

**Comment by klmartinez** [comment]
> 
> 
> The error message suggests that the `INFO/gnom` annotation is not defined in the header. Since you requested multiallelic records to be broken down, this information is needed to properly handle tags defined as `Number=R,A` in the header. Fill the missing information with `bcftools reheader` and it should work. (You'll either need to provide the complete header to the `reheader` command or remove the undefined tags with `annotate -x`.)

Thank you for your reply.

I think I am still a bit confused. Do you mean I need to fill in missing information in the meta-information lines for `##INFO=<ID=gnomAD`? I have 19 INFO/gnom meta-information lines, but all of fields are filled including `Number`, `Type `and `Description`.

For example, one `INFO/gnom` meta-information line looks like the following:
`##INFO=<ID=gnomAD_combined_AF_FIN,Number=A,Type=Float,Description="Allele Frequency among Finnish genotypes, for each ALT allele, in the same order as listed">`

Is there a downside to just removing undefined tags like you suggested? Will I be losing information? Would I just go about doing this with `bcftools annotate -x INFO file.vcf.gz`?

Sorry for all my questions, this is all very new to me and I appreciate any suggestions anyone might have.

**Comment by pd3** [comment]
I see, the problem is then that your VCF is probably truncated at the position `6:30953596` creating a non-sense annotation `gnom`.

**Comment by klmartinez** [comment]
> 
> 
> I see, the problem is then that your VCF is probably truncated at the position `6:30953596` creating a non-sense annotation `gnom`.

I had looked at the rest of the VCF before trying to normalize it, and it looked like sequence data for all chromosomes was present. I had removed the single data point for `6:30953596`, but there were more locations that were causing the same type of error. Would a solution then to be just remove all of this problematic data points? Is there a way to call of them out so I don't have to go through them one by one?
